ls1intum/Artemis
# fix markdown editor italic command

### Checklist

### Motivation and Context


resolved issue #934 

### Description


See pull request #904 description.
In addition some code refactoring to minimize duplicating the code: in the classes bold.command and italic.command only modified selected string by either adding (**) in bold.command or (*) in italic.command.
To remove the markdown we no longer check if text consists the markdown since it could consist it in the middle of selected text not at the beginning and in this case slicing the string would be wrong, instead we check if the border of the selected text is the markdown

### Steps for Testing

1.navigate to course administration
2.edit an existing exercise
3.write text in one of the editors for example in problem statement text editor
4.select text that start or/and ends with white-space
5.click on the"I" (Italic) button
you should see the markdowns only around actual text
6.click preview to see your edited text


### Screenshots


![image](https://user-images.githubusercontent.com/50635005/66831897-05350780-ef59-11e9-86ea-37b2e8ab54e9.png)
![image](https://user-images.githubusercontent.com/50635005/66831921-13832380-ef59-11e9-9d48-bc12aa21bcf1.png)
![image](https://user-images.githubusercontent.com/50635005/66831951-24339980-ef59-11e9-9ea6-6b67c9d43822.png)
